### PR TITLE
DHFPROD-5370:Set correct values for 'processors', 'customHook' and 'headers' when a step is edited

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
Set correct values for 'processors', 'customHook' and 'headers' when a step is edited
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/step/manageMappingStep.sjs [hunk 1]
```diff
@@ -73,6 +87,11 @@ assertions.push(
 hubJsTest.verifyJson(expectedStep, serviceResponse, assertions);
 hubJsTest.verifyJson(expectedStep, stepService.getStep(stepDefinitionType, stepName), assertions);
 
+assertions.push(
+  test.assertEqual("{}", JSON.stringify(serviceResponse.headers)),
+  test.assertEqual("{}", JSON.stringify(serviceResponse.customHook)),
+  test.assertEqual("[]", JSON.stringify(serviceResponse.processors))
+);
```
Comment (bsrikan) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/step/manageMappingStep.sjs:94]: @srinathgit is this assertion required again here?

File: marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings-dialog.tsx [hunk 2]
```diff
@@ -303,12 +303,12 @@ const AdvancedSettingsDialog = (props) => {
         targetDatabase: targetDatabase,
         targetFormat: targetFormat,
         permissions: targetPermissions,
-        headers: parseJSON(headers),
-        processors: parseJSON(processors),
+        headers: isEmptyString(headers) ? {} : parseJSON(headers),
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings-dialog.tsx:306]: We have an RTL test that sets these to empty string and verifies that 'Invalid JSON' error is not thrown. Should that be changed?

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/step/manageMappingStep.sjs (hunk 1 above)
Comment (srinathgit) [reply marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/data-services/step/manageMappingStep.sjs:94]: Yeah, I am calling saveStep() again. So, it ensures the values are present after that as well.

File: marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings-dialog.tsx (hunk 2 above)
Comment (srinathgit) [reply marklogic-data-hub-central/ui/src/components/advanced-settings/advanced-settings-dialog.tsx:306]: No, that test can still remain. If user chooses to set these to empty string, we convert them to valid json values.

**Pull request merged**